pyiceberg/io/pyarrow.py

```diff
@@ -1769,27 +1769,33 @@ def data_file_statistics_from_parquet_metadata(
 
 
 def write_file(io: FileIO, table_metadata: TableMetadata, tasks: Iterator[WriteTask]) -> Iterator[DataFile]:
-    schema = table_metadata.schema()
-    arrow_file_schema = schema.as_arrow()
     parquet_writer_kwargs = _get_parquet_writer_kwargs(table_metadata.properties)
-
     row_group_size = PropertyUtil.property_as_int(
         properties=table_metadata.properties,
         property_name=TableProperties.PARQUET_ROW_GROUP_SIZE_BYTES,
         default=TableProperties.PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT,
     )
 
     def write_parquet(task: WriteTask) -> DataFile:
+        table_schema = task.schema
+        arrow_table = pa.Table.from_batches(task.record_batches)
+        # if schema needs to be transformed, use the transformed schema and adjust the arrow table accordingly
+        # otherwise use the original schema
+        if (sanitized_schema := sanitize_column_names(table_schema)) != table_schema:
+            file_schema = sanitized_schema
+            arrow_table = to_requested_schema(requested_schema=file_schema, file_schema=table_schema, table=arrow_table)
+        else:
+            file_schema = table_schema
```

Comment:
@Fokko wdyt of this? only used the transformed schema when necessary

Reply:
another option could be to push this logic up to the caller of `write_file` or maybe even in `WriteTask`

Reply:
My preference is to leave out all this logic when writing Parquet files. It doesn't affect Parquet files (since the characters are supported). With Iceberg we resolve the fields by ID, so names are unimportant. I would like to get @rdblue's opinion since he did the initial work (I also have a meeting with him tomorrow, and I can ping him there).

Reply:
sounds good! I opened [iceberg/#10120](https://github.com/apache/iceberg/issues/10120) to track this

Reply:
I pinged Ryan and he's in favor of adding the aliases 👍